### ISSUE TITLE
Add NYC TreesCount dataset

### ISSUE THREAD

**Issue by ethanwhite** [weecology/retriever#702]
Just came across this really cool citizen science project via twitter with ~500,000 identified and measured trees in NYC.

Live map:

https://tree-map.nycgovparks.org/

Citizen science project:

https://www.nycgovparks.org/trees/treescount

Data:

https://data.cityofnewyork.us/Environment/2015-Street-Tree-Census-Tree-Data/pi5s-9p35?category=Environment&view_name=2015-Street-Tree-Census-Tree-Data

Link to csv download:

https://data.cityofnewyork.us/api/views/etu3-vwwr/rows.csv?accessType=DOWNLOAD

To really do this right we'll want to parse the first column, which is currently formatted as "POINT (-73.78752645502483 40.73216525220126)" into separate longitude and latitude columns.

**Comment by henrykironde** [comment]
What is the size of the file?

**Comment by henrykironde** [comment]
Okey seems like the first thousand and so lines are empty.

**Comment by ethanwhite** [comment]
Yeah, I saw that too, but there's definitely real data eventually. Anyway, this isn't a priority, just something cool to add at some point.

**Comment by kvnamipara** [comment]
I am unable to open csv file. error is showing as below.
![screen shot 2017-03-03 at 10 48 48 pm](https://cloud.githubusercontent.com/assets/18012563/23561284/a2e2deea-0063-11e7-874d-91711d3c137a.png)

**Comment by ethanwhite** [comment]
Interesting, when I go back through the data page and get the link it is slightly different than the one I originally grabbed: https://data.cityofnewyork.us/api/views/5rq2-4hqu/rows.csv?accessType=DOWNLOAD

I wonder if what role the `5rq2-4hqu` is playing. Is it some sort of randomly changing session type code, in which case this would make things more difficult, or did something just happen to change in the last 4 months.

**Comment by ethanwhite** [comment]
@kvnamipara - thanks for looking into this. I think it would be great to go ahead and develop a JSON file using the above link and then we can dig into why that ID might be changing.

**Comment by ethanwhite** [comment]
I have a query in with the person I know that's involved with the project to find out more about that ID.

**Comment by kvnamipara** [comment]
@ethanwhite I am working on this issue. If you got reply from the person, please let me know.
thanks.

**Comment by kvnamipara** [comment]
@ethanwhite  I have made json file, but i am facing issues in testing. Please help me, I have already checked developer's guide in website. Do i have to test on each, mysql, postgresql?

**Comment by henrykironde** [comment]
@kvnamipara could you put in a PR for review. We can discus more using what you have done

**Comment by ethanwhite** [comment]
I heard back from the NYC TreeCount folks and the link should be stable now.

**Comment by henrykironde** [comment]
@ethanwhite does this have a citation, cant seem to find away to cite this data

**Comment by kvnamipara** [comment]
@ethanwhite @henrykironde 
for some table we are mentioning schema and for some tables we are not. is there any particular criteria based on which we are writing schema?

**Comment by ethanwhite** [comment]
@kvnamipara - the Retriever can auto-detect schemas in a lost of cases, so in the past we've generally added the schemas when we needed to for some reason. Examples of when this was the case included when headers were missing, when we wanted to change column names, or when there was something complicated happening with data types. That said, now that we have moved over to the datapackage standard I think there's some real value in including the schemas because it allows datapackage tools, and potentially us, to provide information about what columns and types are present in the datasets without having to scan them first. So, my preference at the moment would be to include them anytime we have the time and energy to do so.

**Comment by ethanwhite** [comment]
@henrykironde - I asked my contact there. Will report back the preferred citation if/when I hear something.

**Comment by henrykironde** [comment]
Thanks Ethan, @kvnamipara, Sorry I forgot about the reply. But Ethan has explained it well.  I will go ahead and create an issue to include an explanation on the docs

**Comment by ethanwhite** [comment]
The link in my March 3rd comment (https://data.cityofnewyork.us/api/views/5rq2-4hqu/rows.csv?accessType=DOWNLOAD) is stable now in case anyone wants to proceed with this.

**Comment by henrykironde** [comment]
If am not mistaken the url is the same 
https://data.cityofnewyork.us/api/views/5rq2-4hqu/rows.csv?accessType=DOWNLOAD

as in the comment above 
```
The link in my March 3rd comment (
https://data.cityofnewyork.us/api/views/5rq2-4hqu/rows.csv?accessType=DOWNLOAD) 
```

This works well in python2 but not python 3.

**Comment by ethanwhite** [comment]
Yeah that's definitely right. I forgot that we'd actually implemented this.

Closing since this was merged in 72b7c3107abc919dd185136b35bc9bf440feb353.
